pkg/storage/etcd/etcd_watcher.go

```diff
@@ -398,7 +398,9 @@ func (w *etcdWatcher) sendModify(res *etcd.Response) {
 	var oldObj runtime.Object
 	if res.PrevNode != nil && res.PrevNode.Value != "" {
 		// Ignore problems reading the old object.
-		if oldObj, err = w.decodeObject(res.PrevNode); err == nil {
+		node := *res.PrevNode
+		node.ModifiedIndex = res.Node.ModifiedIndex
```

Comment:
This will set the resource version in [APIObjectVersioner.UpdateObject()](https://github.com/hongchaodeng/kubernetes/blob/b9745999c93e21b904aab5dc31a6f24795c1d381/pkg/storage/etcd/api_object_versioner.go#L42-L42)